### PR TITLE
issue-193: Add Dynamo DB Table for SWOT Prior Lakes

### PR DESCRIPTION
**Pull request by torimcd**
Github Issue: #193 Add Dynamo DB Table for SWOT Prior Lakes

### Description

Add new hydrocron dynamoDB table for SWOT prior lakes data

### Overview of work done

* Add code to handle unpacking the prior lakes shapefiles
* update terraform to create additional table
* add unit tests and new test data for lakes

### Overview of verification done

Local unit tests pass and data is loaded from test lakes shapefile into local dynamodb instance

### Overview of integration done

Integration tests revealed that we cannot load large lake items into dynamodb as-is, opened #210 to work that issue. Some smaller lake features were successfully loaded into the new prior-lake table in SIT, so considering this PR complete.

## PR checklist:

* [x] Linted
* [x] Updated unit tests
* [x] Updated changelog
* [x] Integration testing

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

### CODE REVIEW

File: hydrocron/db/io/swot_shp.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def read_shapefile(filepath, obscure_data, columns, s3_resource=None):
         shp_file = gpd.read_file('zip://' + filepath)
         with zipfile.ZipFile(filepath) as archive:
             shp_xml_tree = ET.fromstring(archive.read(filename[:-4] + ".shp.xml"))
+            logging.info(shp_xml_tree)
 
     numeric_columns = shp_file[columns].select_dtypes(include=[np.number]).columns
```
Comment (nikki-t) [inline hydrocron/db/io/swot_shp.py:71]: Can this:

numeric_columns = shp_file[columns].select_dtypes(include=[np.number]).columns

be moved under the if obscure_date statement? Although I think this is trivial, if it is only needed for obscuring data then the column data type look up is only run when obscuring data.

Comment (torimcd) [reply hydrocron/db/io/swot_shp.py]: done

File: hydrocron/db/io/swot_shp.py [hunk 2]
```diff
@@ -66,6 +66,7 @@ def read_shapefile(filepath, obscure_data, columns, s3_resource=None):
         shp_file = gpd.read_file('zip://' + filepath)
         with zipfile.ZipFile(filepath) as archive:
             shp_xml_tree = ET.fromstring(archive.read(filename[:-4] + ".shp.xml"))
+            logging.info(shp_xml_tree)
```
[the remark this answers is not in the document]
Comment (torimcd) [reply hydrocron/db/io/swot_shp.py]: removed

File: hydrocron/db/io/swot_shp.py [hunk 3]
```diff
@@ -76,7 +77,7 @@ def read_shapefile(filepath, obscure_data, columns, s3_resource=None):
             np.random.default_rng().integers(low=2, high=10)*shp_file[numeric_columns],
             shp_file[numeric_columns])
 
-    shp_file = shp_file.astype(str)
+    # shp_file = shp_file.astype(str)
```
[the remark this answers is not in the document]
Comment (torimcd) [reply hydrocron/db/io/swot_shp.py]: done

File: hydrocron/db/load_data.py [hunk 4]
```diff
@@ -247,33 +273,27 @@ def load_data(dynamo_resource, table_name, items):
...
+            logging.warning('Items cannot be parsed, file reader not implemented for table %s', hydrocron_table.table_name)
 
-        else:
-            logging.info("Adding node items to table individually")
-            for item_attrs in items:
-                logging.info("Item node_id: %s", item_attrs['node_id'])
-                hydrocron_table.add_data(**item_attrs)
+    if len(items) > 5:
+        logging.info("Batch adding %s %s items", len(items), feature_name)
+        for i in range(5):
```
[the remark this answers is not in the document]
Comment (frankinspace) [reply hydrocron/db/load_data.py:291]: Ah I see that now, makes sense. Maybe in the previous log message add something like " first five items in batch:" to improve the context?

**Review by frankinspace — APPROVED** [review]

**Review by frankinspace — COMMENTED** [review]
Just noticed one of the tests are failing; can you check on that?

**Pull request merged**